jwst/assign_wcs/tests/test_nirspec.py

```diff
@@ -448,3 +449,114 @@ def test_shutter_size_on_sky():
+    msax, msay, _ = slit2msa(slitx, slity, lam)
+    assert_allclose(slitx, ins_tab['xslitpos'])
+    assert_allclose(slity, ins_tab['yslitpos'])
+    assert_allclose(msax, ins_tab['xmsapos'])
+    assert_allclose(msay, ins_tab['ymaspos'])
+
+    # Coordinates at Collimator exit
+    # Applies the Collimator forward transform to MSa absolute coordinates
+    col = datamodels.open(refs['collimator'])
+    colx, coly = col.model.inverse(msax, msay)
+    col.close()
+    assert_allclose(colx, ins_tab['xcoll'])
+    assert_allclose(coly, ins_tab['ycoll'])
+
+    # After applying direcitonal cosines
+    dircos = trmodels.Unitless2DirCos()
+    xcolDircosi, ycolDircosi, z = dircos(colx, coly)
+    assert_allclose(xcolDircosi, ins_tab['xcolDirCosi'])
+    assert_allclose(ycolDircosi, ins_tab['ycolDirCosi'])
+
+    # MSA to GWA entrance
+    # This runs the Collimator forward, Unitless to Directional cosine, and
+    # 3D Rotation. It uses the corrected GWA tilt value
+    disp = datamodels.DisperserModel(refs['disperser'])
+    disperser = nirspec.correct_tilt(disp, im.meta.instrument.gwa_xtilt,
+                                     im.meta.instrument.gwa_ytilt)
+    collimator2gwa = nirspec.collimator_to_gwa(refs, disperser)
+    x_gwa_in, y_gwa_in, z_gwa_in = collimator2gwa(msax, msay)
+    disp.close()
+    assert_allclose(x_gwa_in, ins_tab['xdispIn'])
+    assert_allclose(y_gwa_in, ins_tab['ydispIn'])
+
+    # Slit to GWA out
+    slit2gwa = slit_wcs.get_transform('slit_frame', 'gwa')
+    x_gwa_out, y_gwa_out, z_gwa_out = slit2gwa(slitx, slity, lam)
+    assert_allclose(x_gwa_out, ins_tab['xdispLaw'])
+    assert_allclose(y_gwa_out, ins_tab['ydispLaw'])
+
+    # CAMERA entrance (assuming direction is from sky to detector)
+    angles = [disperser['theta_x'], disperser['theta_y'],
+              disperser['theta_z'], disperser['tilt_y']]
+    rotation = trmodels.Rotation3DToGWA(angles, axes_order="xyzy",
+                                        name='rotation')
+    dircos2unitless = trmodels.DirCos2Unitless()
+    gwa2cam = rotation.inverse | dircos2unitless
+    x_camera_entrance, y_camera_entrance = gwa2cam(x_gwa_out, y_gwa_out,
+                                                   z_gwa_out)
+    assert_allclose(x_camera_entrance, ins_tab['xcamCosi'])
+    assert_allclose(y_camera_entrance, ins_tab['ycamCosi'])
+
+    # at FPA
+    camera = datamodels.CameraModel(refs['camera'])
+    x_fpa, y_fpa = camera.model.inverse(x_camera_entrance, y_camera_entrance)
+    camera.close()
+    assert_allclose(x_fpa, ins_tab['xfpapos'])
+    assert_allclose(y_fpa, ins_tab['yfpapos'])
+
+    # at SCA These are 0-based , the IDT results are 1-based
+    slit2sca = slit_wcs.get_transform('slit_frame', 'sca')
+    x_sca_nrs1, y_sca_nrs1 = slit2sca(slitx, slity, lam)
+    # At NRS2
+    fpa = datamodels.FPAModel(refs['fpa'])
+    fpa.nrs1_model
+    x_sca_nrs2, y_sca_nrs2 = fpa.nrs2_model.inverse(x_fpa, y_fpa)
+    fpa.close()
```

Comment:
This can be a `with` block.

Also, what does accessing `fpa.nrs1_model` do?  Is that extraneous?